app/controllers/api/v1/admin/posts_controller.rb

```diff
@@ -62,6 +62,12 @@ def unpublish
       updated
     end
 
+    api :POST, '/admin/posts/:post_id/toggle_recommended', 'toggle推荐文章'
+    def toggle_recommended
+      @post.toggle_recommended!
```

Comment:
no need to write a method for that. just use:
```ruby
@post.toggle(:recommended)
```

see: https://apidock.com/rails/ActiveRecord/Base/toggle